dataservice_publisher/app.py

```diff
@@ -44,6 +51,109 @@ async def authenticated(request: web.Request) -> bool:
+        if isinstance(other, MimeType):
+            return self.content_type == other.content_type
+        return False
+
+    def __str__(self) -> str:
+        """Return the mime type as a string."""
+        return self.content_type
+
+
+async def prepare_mime_types(accept_mime_types: List[str]) -> List[str]:
+    """Prepare the accept mime types and sort on q-parameter."""
+    logging.debug(f"Prcoessing accept mime types: {accept_mime_types}")
+    # Assign q-parameter:
+    accept_mime_types_sorted: List[MimeType] = []
+
+    for accept_mime_type in accept_mime_types:
+        mime_type_split = accept_mime_type.split(";")
+        mime_type = MimeType(mime_type_split[0])
+
+        # If q-parameter is present, assign it:
+        for mime_type_part in mime_type_split[1:]:
+            if mime_type_part.startswith("q="):
+                mime_type.q = float(mime_type_part.split("=")[1][0:5])
+
+        accept_mime_types_sorted.append(mime_type)
+
+    # Adjust q-parameters with regard to specificity:
+    # Highest q-parameter is the most specific:
+    for mime_type in accept_mime_types_sorted:
+        logging.debug(f"Ajusting q-parameter for mime type: {mime_type}")
+        if mime_type.content_type.split("/")[0] == "*":
+            pass
+        elif mime_type.content_type.split("/")[1] == "*":
+            mime_type.q = mime_type.q + 0.0001
+        else:
+            mime_type.q = mime_type.q + 0.0002
+
+    # Sort on q-parameter and return list of mime types:
+    accept_mime_types_sorted.sort(key=lambda x: x.q, reverse=True)
+    logging.debug(f"Prcoessing accept mime types sorted: {accept_mime_types_sorted}")
+    return [mime_type.content_type for mime_type in accept_mime_types_sorted]
+
+
+async def decide_content_type(request: web.Request) -> Optional[str]:
+    """Decide the content type of the response."""
+    logging.debug("Deciding content type")
+    if not request.headers.getone(hdrs.ACCEPT, None):
+        content_type = DEFAULT_CONTENT_TYPE["text"]  # pragma: no cover
+    elif "catalogs" in request.path:
+        accept_mime_types: List[str] = (
+            ",".join(request.headers.getall(hdrs.ACCEPT)).replace(" ", "").split(",")
+        )
+        accept_mime_types_sorted = await prepare_mime_types(accept_mime_types)
+        for mime_type in accept_mime_types_sorted:
+            logging.debug(f"Processing mime type: {mime_type}")
+            if mime_type in SUPPORTED_CONTENT_TYPES:
+                content_type = mime_type
+                break
+            elif mime_type == "*/*":
+                content_type = DEFAULT_CONTENT_TYPE["text"]
+                break
+            elif mime_type == "text/*":
+                content_type = DEFAULT_CONTENT_TYPE["text"]
+                break
+            elif mime_type == "application/*":
+                content_type = DEFAULT_CONTENT_TYPE["application"]
+                break
+            else:
+                content_type = None
```

Comment:
Kan være fint å ikke hardkode støtte for `x/*`
```suggestion
            else:
                // Assumes valid mimetypes from `prepare_mime_types`
                mime_maintype, mime_subtype = mime_type.split("/")
                if mime_subtype == "*" and mime_maintype in DEFAULT_CONTENT_TYPE:
                    content_type = DEFAULT_CONTENT_TYPE[mime_maintype]
                    break
                else:
                    content_type = None
```

Reply:
Fiksa i d6e853ebb8450f6135118656449fd29ce2e4a5b0